aiida/orm/nodes/data/code.py

```diff
@@ -120,32 +120,51 @@ def full_label(self):
         """
         return '{}@{}'.format(self.label, self.get_computer_name())
 
+    @property
+    def label(self):
+        """Return the node label.
+
+        :return: the label
+        """
+        return super().label
+
+    @label.setter
+    def label(self, value):
+        """Set the label.
+
+        :param value: the new value to set
+        """
+        if '@' in str(value):
+            msg = "Code labels must not contain the '@' symbol"
+            raise InputValidationError(msg)
+
+        super(Code, self.__class__).label.fset(self, value)
+
     def relabel(self, new_label, raise_error=True):
         """Relabel this code.
 
         :param new_label: new code label
         :param raise_error: Set to False in order to return a list of errors
             instead of raising them.
+
+        .. deprecated:: 1.2.0
+            Will remove raise_error in `v2.0.0`. Use `try/except` instead.
         """
         suffix = '@{}'.format(self.get_computer_name())
         if new_label.endswith(suffix):
             new_label = new_label[:-len(suffix)]
 
-        if '@' in new_label:
-            msg = "Code labels must not contain the '@' symbol"
-            if raise_error:
-                raise InputValidationError(msg)
-            else:
-                return InputValidationError(msg)
-
         self.label = new_label
 
     def get_description(self):
         """Return a string description of this Code instance.
 
         :return: string description of this Code instance
+
+        .. deprecated:: 1.2.0
```

Comment:
Why is this deprecated? Almost all of the ORM classes have `get_description`. If we are going for a property, we should try to be consistent and do this everywhere.

Reply:
I would deprecate this for all classes but didn't want to ask jason to do this in this PR.
Sounds reasonable?

Reply:
I am not sure if we really should replace `get_description` for `description`. The reason that both exist is that `description` is understood to return the value stored in the `description` column and `get_description` is some compound from various columns/attributes. We might be able to find another name for `get_description` to reduce confusion, but in any case I think it should be done in a consistent way across the code and not just in one place

Reply:
Ok, I'll revert this deprecation at some point.